doc/pit/Platform_Integration_Test_high_level_design.md

```diff
@@ -0,0 +1,164 @@
+2022/04/29  V0.2        Clark Lee (guizhao.lh@alibaba-inc.com)  Detail added for platform plugins, RESTful APIs
+
+# Scope
+
+This document describes a subsystem for Platform Integration Test, aka. PIT. PIT system concentrates on standardize and automate white-box switch hardware functionalities verification, along with their driver and firmware. As a result, it will be easier to port SONiC on a new white-box platform which is PIT-verified. Current scope covers all hardware components, such as CPU, memory, SSD, power system, fan, system sensors of various kind, and logical device firmware management, BMC subsystem etc.
+
+# 1. Platform Integration Test (aka. PIT) background
+
+In the blooming SONiC community, lots of users build their software on open hardware（especially white-box switches）. As the devices hardware are built by ODM vendor, not end users (users' SONiC development team), it's necessary to have a systematic method to verify whether the hardware, firmware (such as CPLD, FPGA, BMC, BIOS) and their drivers meet the users' requirement. At this point, PIT system (Platform Integration Test) is introduced for this purpose.
+
+PIT system could lighten up workload for both ODM and end-user because it not only automates hardware, firmware, driver verification efforts, but also it acheive this in a platform independent way, ias it provide a set of platform APIs to unify platform functions, so that hardware from different ODM have a unique view to the users. PIT is the first standard target to simplified SONiC porting on white-box switch in SONiC community. The simplified porting job may attract more and more user and ODM partners to join the SONiC community.
+
+# 2. Platform Integration Test subsystem overview
+
+## 2.1 PIT feature overview
+
+PIT is a sub-system, dedicated for the verification of white-box switch hardware, firmware, and driver.
+PIT provides:
+
+  1) An automation test framework for platform hardware, firmware, and driver test;
+  2) A set of test cases written in platform-independent (CPU, ASIC, product) manner;
+  3) A set of platform APIs to provide hardware function abstraction (complementary to SONiC platform APIs), which help test case logic platform-independent.
+
+## 2.2 PIT deploy scenario
+
+Figure2-1 shows where PIT is used.
+
+![](images/PIT_subsystem_overview.png)
+Figure 2-1
+
+PIT runs in SONiC variant(which is called D4OS, aka. Datacenter Development Diagnostic and Delivery OS, SONiC based, tailored to focus on the developing process of a switch product, co-develop by ODM and users, added users' tools for delivery process, system default configuation, etc). It can be deployed in 3 stage of a switch device developing time axis. Development stage and manufacturing stage which take place in ODM side. Delivery stage which happen on the IDC users. The verified hardware drivers can be added and run in the user SONiC system without changes.
```

Comment:
Please avoid creating new brand names.

Reply:
Hi @lihuay Thanks for the advise. So, should I change back to branch master and redo this PR?

Reply:
I think you can just update this PR also

Reply:
> yes, you need redo the PR against the master branch since the current PR is not for any branches.

Please check this PR, changed the branch to master.
https://github.com/sonic-net/SONiC/pull/1127